### PR TITLE
Issue #2737 - Fulfil NTIA minimum requirements

### PR DESCRIPTION
**Pull request by leec94**
### Addressed Issue

https://github.com/DependencyTrack/dependency-track/issues/2737

### Additional Details

- Takes in supplier and manufacturer fields from an uploaded CycloneDx SBOM. 
- This PR is worked off of @melba-lopez 's fork with some extra changes https://github.com/DependencyTrack/dependency-track/pull/2867 

## Checklist

- [x] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/CONTRIBUTING.md#pull-requests)
- [x] This PR fixes a defect, and I have provided tests to verify that the fix is effective
- [x] This PR implements an enhancement, and I have provided tests to verify that it works as intended
- [ ] This PR introduces changes to the database model, and I have added corresponding [update logic](https://github.com/DependencyTrack/dependency-track/tree/master/src/main/java/org/dependencytrack/upgrade)
- [ ] This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly

### CODE REVIEW

**Comment by leec94** [comment]
can't fork melbas fork (https://github.com/DependencyTrack/dependency-track/pull/2867) so im adding it here for debugging

**Comment by leec94** [comment]
looks like the CI tests aren't running, do i have to change this to not be a draft to run them?

**Comment by nscuro** [comment]
> looks like the CI tests aren't running, do i have to change this to not be a draft to run them?

Workflows on PRs from first time contributors need approval from maintainers first 😅 Approved it just now!

File: src/main/java/org/dependencytrack/model/Component.java
Comment (nscuro) [inline src/main/java/org/dependencytrack/model/Component.java:119]: We need to make the new columns nullable, otherwise there'll be constraint violations on existing deployments (where the columns will be `NULL` for all existing components and projects).

Just adding `allowsNull = "true"` to the `@Column` annotation will do the trick!

**Review by nscuro — CHANGES_REQUESTED** [review]

File: src/main/java/org/dependencytrack/model/Component.java
Comment (leec94) [reply src/main/java/org/dependencytrack/model/Component.java]: made the change, thanks for the feedback!

**Review by nscuro — APPROVED** [review]
Thanks @leec94! 🚀

**Pull request merged**
